lib/ConfigFileProcessor.js

```diff
@@ -0,0 +1,327 @@
+/**
+ * Copyright (c) 2018, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const async = require('async');
+
+const { AppOptionsName, DomainTypes, EntityType, EnvOptionsName, LogLevel, OperationType } = require('./Constants');
+const { Endpoints, getConfigTypeError, getCustomNotFoundError, getObjectByOmitting, isEmpty, isNullOrUndefined, readJSON } = require('./Utils');
+
+class ConfigFileProcessor {
```

Comment:
The way it is, this file will grow a lot with the implementation of more config file features. I would separate it for the different types of configuration - environment, service, etc.

Reply:
Yeah, I agree. I was just thinking of doing the separation when the need arises as then I'd have more clarity how exactly to separate things.

Reply:
That's fine, but we will definitely need to do this before MVP.